### PR TITLE
Feature/update array utils

### PR DESCRIPTION
**Pull request by djkapner**
This PR should merge after #33

NOTE: I don't feel like the code in `rois/array_utils.py` is optimal, and I don't think the test coverage for this new function is complete. But, the basic functionality I want is there, and I want to merge and move on to finishing the pipeline and come back to this when corner cases come up.

Purpose of the function `content_extents`:
* given an ROI object, give back some extents and padding to index and pad a video frame

Example:
image
from:
```
import segmentation_labeling_app.rois.rois as rois
import segmentation_labeling_app.transforms.array_utils as au
import matplotlib.pyplot as plt
import numpy as np

# get an ROI and its mask
roi = rois.ROI.roi_from_query(1972)
mask = roi.generate_ROI_mask()

# make a moview frame with context around the ROI
x = np.arange(512)
z = np.meshgrid(np.sin(x/4), np.cos(x/4))
frame = np.abs(z[0] + z[1]) * 0.4
frame += roi._sparse_coo.toarray()

# use the new function to extract the part of the movie
# frame based on the ROI
e, pad = au.content_extents(roi._sparse_coo, shape=(128, 128), target_shape=frame.shape)
frame_extract = np.pad(frame[e[0]: e[1], e[2]:e[3]], pad)

f, a = plt.subplots(1, 3, clear=True, num=1)
a[0].imshow(mask)
a[0].set_title('mask')
a[1].imshow(frame)
a[1].set_title('movie frame')
a[2].imshow(frame_extract)
a[2].set_title('extracted region of frame')
```

changing the roi placement:
```
roi = rois.ROI.roi_from_query(1972)
roi._sparse_coo.row -= 60
```
image

and again:
```
roi = rois.ROI.roi_from_query(1972)
roi._sparse_coo.row += 400
roi._sparse_coo.col += 370
```
image

### CODE REVIEW

**Comment by kschelonka** [comment]
This hasn't been rebased yet, right? I think the fractional threshold got superseded by quantiles in the prior PR, but I see it here.

**Comment by djkapner** [comment]
> This hasn't been rebased yet, right? I think the fractional threshold got superseded by quantiles in the prior PR, but I see it here.

Thanks, rebased.

File: segmentation_labeling_app/transforms/array_utils.py [hunk 1]
```diff
@@ -44,6 +44,89 @@ def content_boundary_2d(arr: Union[np.ndarray, coo_matrix]) -> np.ndarray:
...
+        (csc_matrix, etc.) as long as they have the toarray() method to
+         convert them to np.ndarray.
+    shape: (Tuple[int,int]) Desired final shape after padding is applied.
+        If smaller than the input array, will return the input array
+        without any changes.
+    target_shape: (Tuple[int, int]) Extent of array to be indexed. If None
+        padding will still handle top and left, but bottom and right padding
+        will always be zero
+
+    Returns
```
Comment (isaak-willett) [inline segmentation_labeling_app/transforms/array_utils.py:66]: Double return doc string, I think it got left by mistake

**Comment by djkapner** [comment]
closing, becoming part of #37
